main.js

```diff
@@ -0,0 +1,220 @@
+//Update Data structure Week2
+const currencyData = [
+    {
+      timestamp: new Date().getTime(),
+      date: new Date(),
+      baseCurrency: "Danish Krone",
+      rates: {}
+    },
+    {
+      timestamp: new Date().getTime(),
+      date: new Date(),
+      baseCurrency: "Euro",
+      rates: {}
+    },
+    {
+      timestamp: new Date().getTime(),
+      date: new Date(),
+      baseCurrency: "Dollar",
+      rates: {}
+    },
+    {
+        timestamp: new Date().getTime(),
+        date: new Date(),
+        baseCurrency: "Swedish Krone",
+        rates: {}
+    },
+    {
+        timestamp: new Date().getTime(),
+        date: new Date(),
+        baseCurrency: "Norwegian Krone",
+        rates: {}
+      },
+      {
+        timestamp: new Date().getTime(),
+        date: new Date(),
+        baseCurrency: "Swiss Frank",
+        rates: {}
+      },
+      {
+        timestamp: new Date().getTime(),
+        date: new Date(),
+        baseCurrency: "Pound Sterling",
+        rates: {}
+      },
+      {
+        timestamp: new Date().getTime(),
+        date: new Date(),
+        baseCurrency: "Turkish Lira",
+        rates: {}
+      },      
+
+  ];
+  
+//Function to add currency rates to base currencies  
+function addCurrency() {
+    //Retrieve values from inputs
+    const baseCurrency = document.getElementById('baseCurrency').value.trim();
+    const conjugateCurrency = document.getElementById('conjugateCurrency').value;
+    const rate = parseFloat(document.getElementById('rate').value);
+  
+    //Find the index of the selected base currency
+    const baseCurrencyIndex = currencyData.findIndex(currency => currency.baseCurrency === baseCurrency);
+  
+    if (baseCurrencyIndex !== -1) {
+      if (!currencyData[baseCurrencyIndex].rates) {
```

Comment:
But if I delete this, would I refactor my object structure?

Reply:
It is related with `findIndex` issue :(

Reply:
you can keep it then, just please fix the `searchCurrency` so it meets the assignment 
```
implement a function to find a specific currency rate, searching by the currency from or to.
Basically, it is just a search function! With a twist, though, because you have both the from and the to fields. As a user, I want to find a specific rate, but I am too lazy to scan through all the rates with my eyes, I want to be able to type in and search!
```